test/performance/ThroughputSubscriber.cpp

```diff
@@ -190,85 +194,91 @@ void ThroughputSubscriber::CommandSubListener::onNewDataMessage(Subscriber* subs
                         DynamicTypeBuilderFactory::get_instance()->create_uint32_type());
                     struct_type_builder->add_member(1, "data",
                         DynamicTypeBuilderFactory::get_instance()->create_sequence_builder(
-                            DynamicTypeBuilderFactory::get_instance()->create_byte_type(), m_up.m_datasize
+                            DynamicTypeBuilderFactory::get_instance()->create_byte_type(), m_datasize
                         ));
                     struct_type_builder->set_name("ThroughputType");
 
-                    m_up.m_pDynType = struct_type_builder->build();
-                    m_up.m_DynType.CleanDynamicType();
-                    m_up.m_DynType.SetDynamicType(m_up.m_pDynType);
+                    m_pDynType = struct_type_builder->build();
+                    m_DynType.CleanDynamicType();
+                    m_DynType.SetDynamicType(m_pDynType);
 
-                    Domain::registerType(m_up.mp_par, &m_up.m_DynType);
+                    Domain::registerType(mp_par, &m_DynType);
 
-                    m_up.mp_datasub = Domain::createSubscriber(m_up.mp_par, m_up.subAttr, &m_up.m_DataSubListener);
-
-                    m_up.m_DynData = DynamicDataFactory::get_instance()->create_data(m_up.m_pDynType);
+                    m_DynData = DynamicDataFactory::get_instance()->create_data(m_pDynType);
                 }
                 else
                 {
-                    delete(m_up.throughputin);
-                    //m_up.throughputin = nullptr;
-                    m_up.throughputin = new ThroughputType((uint16_t)m_up.m_datasize);
+                    delete(throughput_t);
```

Comment:
Shouldn't we check nullptr before deleting?

Reply:
I think isn't necessary:

`delete(nullptr)` is defined to do nothing. About potential double free: In the constructor is set to `nullptr`. In line 555 is set to `nullptr` after deleting it. So the only deletion, if I'm not wrong, is performed in that line is when reserved at line 214.
So double free shouldn't happen.